src/ui/widgets.rs

```diff
@@ -0,0 +1,85 @@
+use bevy::{ecs::system::EntityCommands, prelude::*};
+
+use super::{BUTTON_NORMAL, BUTTON_TEXT, LABEL_TEXT};
+
+/// Helper trait for creating common widgets.
+pub trait MyWidgets<'w> {
+    /// Spawns a simple root node.
+    fn my_root(&mut self) -> EntityCommands;
+
+    /// Spawns a simple button node.
+    fn my_button<I: Into<String>>(&mut self, text: I) -> EntityCommands;
+
+    /// Spawns a simple label.
+    fn my_label<I: Into<String>>(&mut self, text: I) -> EntityCommands;
+}
```

Comment:
Bikeshedding: Rename to `SpawnWidgetsExt`, `spawn_root`, `spawn_button`, `spawn_label`?

Reply:
I want this to be very clearly user defined, so im straying away from standard bevy command names.

Reply:
I'd leave it as-is for this PR, then go talk to some folks in the ui-dev channel on Discord about what they'd do. I'm sure that there's a bunch of people there that have way more experience with how to organize vanilla Bevy UI than any of us. We should learn from them :)